components/client/src/test/unit/java/com/hotels/styx/client/OriginsInventoryTest.java

```diff
@@ -85,207 +86,397 @@ public void stop() {
         logger.stop();
     }
 
+    /*
+     * Setting the origins
+     */
     @Test
-    public void addingANewOriginWillAddToActiveSetAndInitiateTheMonitoring() {
-        inventory.addOrigins(ORIGIN);
+    public void setNewOrigins() {
+        inventory.setOrigins(ORIGIN_1, ORIGIN_2);
+
+        assertThat(inventory, hasActiveOrigins(2));
+        verify(monitor).monitor(singleton(ORIGIN_1));
+        verify(monitor).monitor(singleton(ORIGIN_2));
+
+        assertThat(gaugeValue("origins.generic-app.app-01.status"), isValue(1));
+        assertThat(gaugeValue("origins.generic-app.app-02.status"), isValue(1));
+
+        verify(eventBus).post(any(OriginsInventorySnapshot.class));
+    }
+
+    @Test
+    public void updatesOriginPortNumber() throws Exception {
```

Comment:
This name is also not as informative as it could be. Perhaps something like "replaces origin when port number has changed"?

Reply:
Which is a longwinded way of saying "updates origin port number" ;-)